core/src/main/java/com/yahoo/imapnio/async/client/ImapAsyncSessionClientContext.java

```diff
@@ -0,0 +1,32 @@
+package com.yahoo.imapnio.async.client;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class for allowing client putting context about this session.
+ */
+public final class ImapAsyncSessionClientContext {
```

Comment:
we do not need the class, we can pass in an object, and the toString can print whatever data the client wants,